### PR TITLE
Improve Module docs: keep same order in Module's @callback example

### PR DESCRIPTION
**Pull request by eksperimental**
Order should be the same when defining the callback as when implementing it

### CODE REVIEW

**Comment by whatyouhide** [comment]
I approved the changes but don't like the commit message with the : in it, I would rephrase as "Reorder some code around @callback in the docs for Module" or something like this.

**Comment by eksperimental** [comment]
@whatyouhide: how about "Reorder code in Module's @callback example"

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:
